kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveStore.java

```diff
@@ -138,7 +137,8 @@ public void init(final StateStoreContext context, final StateStore root) {
       registration = new ResponsiveStoreRegistration(
           name.kafkaName(),
           topicPartition,
-          offset == -1 ? 0 : offset
+          offset == -1 ? 0 : offset,
+          buffer::flush
```

Comment:
we probably don't want to actually flush every commit and instead replicate the behavior in `StateManagerUtil#checkpointNeeded` (and make it configurable). Based on my testing, if we flush every 100ms we're going to be in a bit of a pickle! especially when we explode partitions since we'll be sending an offset write to a partition even if there are no data writes to that partition